### PR TITLE
follow the person's followers with username

### PR DESCRIPTION
**Pull request by prateek-ghanghas**
# what PR do?
1. there are two functions, one is "follow_persons_followers()" , it fetches the follower's ids by providing username. And the other is "follow_a_person()", it follows the person one by one. We time sleep for 5 sec. before every  function "follow_a_person()" calling . when the follow limit exhaust, we time sleep for 3600 sec.

### CODE REVIEW

File: product_hunt_utils.py [hunk 1]
```diff
@@ -1,84 +1,131 @@
...
+    '_ga_WZ46833KH9': 'GS1.1.1691513462.24.1.1691513932.60.0.0',
+}
+
 headers = {
-            'authority': 'www.producthunt.com',
-            'accept': '*/*',
-            'accept-language': 'en-US,en;q=0.9',
-            'content-type': 'application/json',
-            'origin': 'https://www.producthunt.com',
-            'referer': 'https://www.producthunt.com/@cyriljeet',
```
Comment (manmohitgrewal1) [inline product_hunt_utils.py:36]: check if the request works without the "referer" header. having a specific URL in refer does feel good

File: product_hunt_utils.py [hunk 2]
```diff
@@ -1,84 +1,131 @@
...
+    'sec-ch-ua-platform': '"Windows"',
+    'sec-fetch-dest': 'empty',
+    'sec-fetch-mode': 'cors',
+    'sec-fetch-site': 'same-origin',
+    'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36',
+    'x-requested-with': 'XMLHttpRequest',
+}
 
-def get_commentor_info_for_product_id(product_id: int)-> list:
+def follow_persons_followers(person_username):
```
Comment (manmohitgrewal1) [inline product_hunt_utils.py:47]: add param and return type

File: product_hunt_utils.py [hunk 3]
```diff
@@ -1,84 +1,131 @@
...
-        This function try follows a person by providing id of that person
-        and return True if succeed otherwise False
-        :param person_id: sample: 922865
-        :return True or False
-        """
-
-        json_data = {
+
+
+def follow_a_person(person_id):
```
Comment (manmohitgrewal1) [inline product_hunt_utils.py:126]: add doc string

Comment (manmohitgrewal1) [reply product_hunt_utils.py:126]: add param and return type

File: product_hunt_utils.py [hunk 4]
```diff
@@ -1,84 +1,131 @@
...
+    'sec-ch-ua-mobile': '?0',
+    'sec-ch-ua-platform': '"Windows"',
+    'sec-fetch-dest': 'empty',
+    'sec-fetch-mode': 'cors',
+    'sec-fetch-site': 'same-origin',
+    'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36',
+    'x-requested-with': 'XMLHttpRequest',
+}
 
-def get_commentor_info_for_product_id(product_id: int)-> list:
```
Comment (manmohitgrewal1) [inline product_hunt_utils.py:47]: where is `get_commentor_info_for_product_id` function?

**Comment by manmohitgrewal1** [comment]
Never use special characters in while naming any branch, file, an directory. `follow_person"s_followers` should be `follow_person_followers` 
You can still fix this by creating a new branch over `follow_person"s_followers`. You can do so by checking out `follow_person"s_followers` and then executing `git checkout -b  prateek/add_function_to_follow_followers_of_username`. Then create a PR with base `main`. This way we will abandon `follow_person"s_followers`

File: product_hunt_utils.py (hunk 4 above)
Comment (prateek-ghanghas) [reply product_hunt_utils.py:47]: actually i did not  pushed whole code , just the following part

**Pull request closed without merging**